src/Video.tsx

```diff
@@ -512,41 +512,51 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
       [onControlsVisibilityChange],
     );
 
-    const usingExternalGetLicense = drm?.getLicense instanceof Function;
+    const selectedDrm = source?.drm || drm;
+    const usingExternalGetLicense = selectedDrm?.getLicense instanceof Function;
 
     const onGetLicense = useCallback(
       async (event: NativeSyntheticEvent<OnGetLicenseData>) => {
         if (!usingExternalGetLicense) {
           return;
         }
         const data = event.nativeEvent;
-        let result;
-        if (data?.spcBase64) {
-          try {
-            // Handles both scenarios, getLicenseOverride being a promise and not.
-            const license = await drm.getLicense(
-              data.spcBase64,
-              data.contentId,
-              data.licenseUrl,
+        try {
+          if (!data?.spcBase64) {
+            throw new Error('No spc received');
+          }
+          // Handles both scenarios, getLicenseOverride being a promise and not.
+          const license = await Promise.resolve(selectedDrm.getLicense(
+            data.spcBase64,
+            data.contentId,
+            data.licenseUrl,
+            data.loadedLicenseUrl,
+            )
+          ).catch(() => {
+            throw new Error('fetch error');
+          });
+          if (typeof license !== 'string') {
+            throw Error('Empty license result');
+          }
+          if (nativeRef.current) {
+            NativeVideoManager.setLicenseResultCmd(
+              getReactTag(nativeRef),
+              license,
+              data.loadedLicenseUrl,
+            );
+          }
+        } catch (e) { 
+          const msg = e instanceof Error ? e.message : 'fetch error';
```

Comment:
It would be better to avoid hard coded string here, but I approve anyway.
There is a lack of error code spec/management in the package...